bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Rule.java

```diff
@@ -101,6 +104,30 @@ public String getUID() {
     }
 
     /**
+     * Gets the scope of the Rule, if any exist. This property is either set by the {@link RuleEngine} when
+     * the {@link Rule} is added or by the creating party. It's an optional property.
+     * <br/><br/>Implementation note
+     *    <br/>The scope is part of the {@code UID} and the prefix thereof.
+     *    <br/>If the UID does not contain a {@link SCOPE_DELIMITER} {@code null} will be returned.
+     *    <br/>If the UID does contain a {@link SCOPE_DELIMITER} the prefix until the first occurrence will be returned.
+     *    <br/>If the prefix would have a zero length {@code null} will be returned.
+     *
+     * @return scope of this {@link Rule}, or {@code null} if no scope or an empty scope is found
+    */
+    public String getScope() {
```

Comment:
I think the scope is not part of the rule's features. It is up to your business logic how to treat uid of rule. What happen if  I don't like to divide my rule into scopes or I want to divide  them by groups or by color, in that case should I use your getScope() method to retrieve my rules?  I think we have agreed to add a method in ruleRegistry.getAll(Predicate) which can be used by your business logic and get scopes, groups and etc.